src/app/detail/ui/MenuBox.tsx

```diff
@@ -0,0 +1,56 @@
+'use client'
+
+import { ListBtn } from '@/components/buttons/ListBtn'
+import { MoreBtn } from '@/components/buttons/MoreBtn'
+import { Box, Popper } from '@mui/material'
+import { useState } from 'react'
+import { ClickAwayListener } from '@mui/base'
+import store from '@/redux/store'
+import { setShowConfirmDeleteModal } from '@/redux/features/taskDetailsSlice'
+
+export const MenuBox = () => {
```

Comment:
is this supposed to be a generic component? It doesn't seem generic given that it has an explicit Delete menu item. The options should probably be passsed in as props right?

Reply:
This is not a generic component. Inside the `ui` folders of every pages, I am placing the client components with `use client` directive that are only useful for that particular page.